docs/tpm_lpc_spi_interface.md

```diff
@@ -0,0 +1,150 @@
+actually is in register, TPM must handle this, it must also handle read from the
+middle of register, e.g. if we have a 32-bit register at 0x20, 1 byte read from
+0x20 must yield the first byte of that register, read from 0x21 must yield the
+second byte from that register.
+
+Software must never attempt to cross register size in a same transfer. In case
+of LPC this won't be noticed by TPM as all multi-byte LPC transfers are split
+into multiple transfers, e.g. if hosts writes 0x11223344 to register 0x10 on LPC
+it will appear as
+- 0x44 write to 0x10
+- 0x33 write to 0x11
+- 0x22 write to 0x12
+- 0x11 write to 0x13
+
+Even if register boundary were crossed this doesn't make any difference for TPM.
+In case of SPI data payload of a single transfer cannot be bigger than size of
+register and cannot cross register boundary. Still, this must be handled, TPM
+specification does not define clearly what the behaviour should be.
+> For SPI, if a TPM receives an access request with a length that exceeds the
+> size of the register specified in the transaction address:
+> (...)
+> b. Writes:
+>   The TPM SHALL update the register designated by the start address.
+>   The TPM MAY update additional, adjacent registers within the targeted
+>   address range.
+>   The TPM SHALL NOT change the state of adjacent registers if the writes to
+>   that register are dropped.
+>
+> The TPM SHOULD NOT abort (as defined in Section 6.5.1.1 Bus Aborts) an entire
+> transaction that crosses a register boundary.
+
+We choosed to write as many registers as possible, if register write is dropped
+(such as when register is not present or read-only) bus abort is raised (see
+section 6.5.1.1). This stops write on the register which triggered error and
+further registers are not updated.
+
+TPM must implement the following registers (non-exauhstive list):
+- TPM_DID_VID_x - contains Vendor and Device ID of the TPM, see **Table 13** for
+  details.
+- TPM_RID_x - contains revision ID, see **Table 14 for details**.
+- TPM_INTERFACE_ID_x - mandatory register, see **Table 15**.
+- interrupt control/status registers
+- other registers are required, **Table 19** for FIFO interface requirements.
+
+
+## TPM commands
+
+There are two interfaces for sending TPM commands and reading back responses,
+one is CRB not covered here and another FIFO. See section 6.5.2 for details
+about FIFO mode.
+
+In FIFO mode command is written to `TPM_DATA_FIFO_x` register. Host writes
+entire command packet to FIFO and triggers command execution by setting
+`tpmGo` in the `TPM_STS` (see **Table 21**) register which triggers command
+execution. When response is available TPM signals it by setting `dataAvail` bit
+in status register. Host can now read response from `TPM_DATA_FIFO_x`.
+
+There is also `TPM_XDATA_FIFO_x` register which allows for writes that span
+multiple bytes, this is especially useful for SPI as it can send more than one
+byte in a single packet.
```

Comment:
```suggestion
dword in a single packet.
```
SPI allowed for 4B packets in `TPM_DATA_FIFO_x` already, this potentially can allow for even more.

Reply:
@krystian-hebel Fixed.
TPM_XDATA_FIFO allows for transfers up to 64 bytes, however I'm not sure how that would be implemented. Since address is incremented after each byte TPM_XDATA_FIFO would have to span 64 bytes in address space, but according to spec it is 4 bytes long, followed by reserved region.

Reply:
![Zrzut ekranu z 2022-06-23 08-23-58](https://user-images.githubusercontent.com/44615573/175229783-71667a96-d56f-4cd1-8eff-d936331a03c5.png)

Reply:
I added more info from our conversation into document